File: RazorPages/Reservation.cs
public class Reservation
{
    public Room Room { get; set; }
    public DateTime DateTime { get; set; }
    public string ReservedBy { get; set; }

    public Reservation(Room room, DateTime dateTime, string reservedBy)
    {
        Room = room;
        DateTime = dateTime;
        ReservedBy = reservedBy;
    }

    public override string ToString()
    {
        return $"Room: {Room.RoomId} - {Room.RoomName}, Date: {DateTime.ToShortDateString()}, Time: {DateTime.ToShortTimeString()}, Reserved By: {ReservedBy}";
    }
}

File: RazorPages/WebApp/wwwroot/js/reservation.js
// rooms.js

function validateForm() {
    var roomName = document.getElementById("roomName").value;
    var reservationDate = document.getElementById("reservationDate").value;

    // Check if roomName is selected
    if (roomName.trim() == "") {
        alert("Please select a room.");
        return false;
    }

    // Check if reservationDate is selected
    if (reservationDate.trim() == "") {
        alert("Please select a reservation date and time.");
        return false;
    }

    // You can add more validation logic here as needed

    return true;
}
